### PR TITLE
Feture/domain lombok changes

### PR DESCRIPTION
**Pull request by RVelp1**

### CODE REVIEW

File: cerberus-domain/src/main/java/com/nike/cerberus/domain/SecureFileCurrent.java [hunk 1]
```diff
@@ -33,93 +37,46 @@
...
     this.data = data != null ? Arrays.copyOf(data, data.length) : null;
-    return this;
-  }
-
-  public int getSizeInBytes() {
-    return sizeInBytes;
-  }
-
-  public SecureFileCurrent setSizeInBytes(int sizeInBytes) {
+    ;
```
Comment (mayitbeegh) [inline cerberus-domain/src/main/java/com/nike/cerberus/domain/SecureFileCurrent.java]: This is probably not needed

File: cerberus-domain/src/main/java/com/nike/cerberus/domain/SecureFileVersion.java [hunk 2]
```diff
@@ -33,103 +37,47 @@
...
-  }
+  public static class SecureFileVersionBuilder {
+    private byte[] data;
 
-  public SecureFileVersion setActionTs(OffsetDateTime actionTs) {
-    this.actionTs = actionTs;
-    return this;
+    public SecureFileVersionBuilder data(byte[] data) {
+      this.data = Arrays.copyOf(data, data.length);
+      ;
```
Comment (mayitbeegh) [inline cerberus-domain/src/main/java/com/nike/cerberus/domain/SecureFileVersion.java]: This is probably not needed

File: lombok.config [hunk 3]
```diff
@@ -1,2 +1,3 @@
 config.stopBubbling = true
 lombok.addLombokGeneratedAnnotation = true
+lombok.anyconstructor.addconstructorproperties = true
```
Comment (mayitbeegh) [inline lombok.config]: Is this needed?

File: cerberus-domain/src/test/java/com/nike/cerberus/validation/UniqueOwnerValidatorTest.java [hunk 4]
```diff
@@ -41,53 +43,60 @@ public void setup() {
...
 
     Assert.assertTrue(subject.isValid(safeDepositBox1, mockConstraintValidatorContext));
     Assert.assertTrue(subject.isValid(safeDepositBox2, mockConstraintValidatorContext));
   }
 
   @Test
   public void owner_in_group_permissions_is_invalid() {
-    UserGroupPermission userGroupPermission = new UserGroupPermission();
+    UserGroupPermission userGroupPermission = UserGroupPermission.builder().name("owner").build();
     userGroupPermission.setName("owner");
```
Comment (mayitbeegh) [inline cerberus-domain/src/test/java/com/nike/cerberus/validation/UniqueOwnerValidatorTest.java]: This line can be deleted

File: cerberus-domain/src/main/java/com/nike/cerberus/domain/UserCredentials.java [hunk 5]
```diff
@@ -17,24 +17,26 @@
...
 
 import java.util.Arrays;
+import lombok.Getter;
+import lombok.Setter;
 
 /** Represents the user's credentials sent during authentication. */
 public class UserCredentials {
 
-  private final String username;
+  @Getter @Setter private String username;
```
Comment (mayitbeegh) [inline cerberus-domain/src/main/java/com/nike/cerberus/domain/UserCredentials.java]: I think UserCredentials should avoid using @Setter. It's intended to never change its state after init.

File: cerberus-domain/src/main/java/com/nike/cerberus/domain/UserCredentials.java [hunk 6]
```diff
@@ -17,24 +17,26 @@
...
 
-  private final byte[] password;
+  private byte[] password;
 
-  public UserCredentials(final String username, final byte[] password) {
+  public UserCredentials(String username, byte[] password) {
```
Comment (mayitbeegh) [inline cerberus-domain/src/main/java/com/nike/cerberus/domain/UserCredentials.java]: Let's add final back to these parameters

File: lombok.config (hunk 3 above)
Comment (RVelp1) [reply lombok.config]: removed

File: cerberus-domain/src/main/java/com/nike/cerberus/domain/UserCredentials.java (hunk 6 above)
Comment (RVelp1) [reply cerberus-domain/src/main/java/com/nike/cerberus/domain/UserCredentials.java]: added

File: cerberus-domain/src/test/java/com/nike/cerberus/validation/UniqueOwnerValidatorTest.java (hunk 4 above)
Comment (RVelp1) [reply cerberus-domain/src/test/java/com/nike/cerberus/validation/UniqueOwnerValidatorTest.java]: deleted

File: cerberus-domain/src/main/java/com/nike/cerberus/domain/UserCredentials.java (hunk 5 above)
Comment (RVelp1) [reply cerberus-domain/src/main/java/com/nike/cerberus/domain/UserCredentials.java]: made changes

**Review by mayitbeegh — APPROVED** [review]

**Comment by mayitbeegh** [comment]
There seems to be 3 failed unit tests

**Pull request closed without merging**